src/JunctionTree.jl

```diff
@@ -756,6 +756,16 @@ function initTreeMessageChannels!(tree::MetaBayesTree)
   return nothing
 end
 
+function refTreeMessageChannelsFromBTND!(tree::BayesTree)
+  for inci in tree.bt.inclist
+    for e in inci
+        downMsg = e.target.data.dwnMsgChannel
+        upMsg = e.target.data.upMsgChannel
+        push!(tree.messages, e.index=>(upMsg=upMsg,downMsg=downMsg))
+    end
+  end
+  return nothing
+end
```

Comment:
Based on https://github.com/JuliaRobotics/IncrementalInference.jl/issues/675#issuecomment-679954452
A container that keeps all channels in one place is also handy for debugging and monitoring. we can also send termination messages to cleanly exit as with PCSM